libs/vkd3d/state.c

```diff
@@ -3652,33 +3754,52 @@ HRESULT vkd3d_bindless_state_init(struct vkd3d_bindless_state *bindless_state,
             VKD3D_BINDLESS_SET_SAMPLER, VK_DESCRIPTOR_TYPE_SAMPLER)))
         goto fail;
 
-    if (FAILED(hr = vkd3d_bindless_state_add_binding(bindless_state, device,
-            VKD3D_BINDLESS_SET_CBV | extra_bindings,
-            vkd3d_bindless_state_get_cbv_descriptor_type(bindless_state))))
-        goto fail;
+    if (vkd3d_bindless_supports_mutable_type(device))
+    {
+        INFO("Device supports VK_VALVE_mutable_descriptor_type.\n");
+        /* If we can, prefer to use one universal descriptor type which works for any descriptor. */
+        bindless_state->flags |= VKD3D_BINDLESS_MUTABLE_TYPE;
```

Comment:
I'd rather have this check + setting flags inside `vkd3d_bindless_state_get_bindless_flags`, and `if (flags & VKD3D_BINDLESS_MUTABLE_TYPE)` here instead.

Reply:
Fixed.

Reply:
actually, derp, now I remember why I did the check there and not in get_bindless_flags ...

Reply:
Fixed.